packages/amplify-graphql-relational-transformer/src/belongs-to/belongs-to-directive-transformer-factory.ts

```diff
@@ -21,18 +21,22 @@ export const getBelongsToDirectiveTransformer = (
     case 'POSTGRES':
       return belongsToDirectivePostgresTransformer;
     case 'DYNAMODB':
-      // If `references` are passed to the directive, we'll use the references relational
+    // If references are passed to the directive, we'll use the references relational
       // modeling approach.
-      if (config.references?.length >= 1) {
-        // Passing both `references` and `fields` is not supported.
-        if (config.fields?.length > 0) {
+      if (config.references) {
+        // Passing both references and fields is not supported.
+        if (config.fields) {
           // TODO: Better error message
           throw new Error('Something went wrong >> cannot have both references and fields.');
         }
+        if (config.references.length < 1) {
+          throw new Error(`Invalid @belongsTo directive on ${config.field.name.value} - empty references list`);
```

Comment:
This is I suppose a breaking change introduced in gen2?

Reply:
Why would this be a breaking change? This is essentially confirming that a relational directive doesn't specific an empty `references` argument. i.e. `@hasMany(references: [])`. The `fields` based behavior of `@hasMany` is unchanged.

Reply:
Sorry, I mis-understood this to be failing if `references` is not specified on belongsTo, but looks like we support both `references` and `fields`, so are backwards compatible.